apps/qubit/modules/digitalobject/actions/showComponent.class.php

```diff
@@ -74,12 +74,19 @@ public function execute($request)
      */
     private function checkShowGenericIcon()
     {
+        $resourceObject = $this->resource->object;
+
+        // For digital objects, get the parent resouce (IO or Actor) for Acl check
+        if (!$resourceObject && ('QubitDigitalObject' === get_class($this->resource))) {
+            $resourceObject = $this->resource->parent->object;
+        }
+
         switch ($this->usageType) {
             case QubitTerm::REFERENCE_ID:
-                return !QubitAcl::check($this->resource->object, 'readReference');
+                return !QubitAcl::check($resourceObject, 'readReference');
 
             case QubitTerm::THUMBNAIL_ID:
-                return !QubitAcl::check($this->resource->object, 'readThumbnail');
+                return !QubitAcl::check($resourceObject, 'readThumbnail');
         }
```

Comment:
Based on this switch it will always be the resource parent, unless `$this->usageType` doesn't match `$this->resource->usageId`. Only the master will be related to the IO/Actor.